### ISSUE TITLE
imports incorrectly marked as unused when augmenting the module it was imported from in `.d.ts` file

### ISSUE THREAD

**Issue by DetachHead** [microsoft/TypeScript#54042]
# Bug Report

### 🔎 Search Terms
unused import augment module

### 🕗 Version & Regression Information
5.1.0-dev.20230427
### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?noUncheckedIndexedAccess=true&noUnusedLocals=true&target=99&jsx=0&resolvePackageJsonExports=true&useUnknownInCatchVariables=true&exactOptionalPropertyTypes=true&noFallthroughCasesInSwitch=true&noImplicitOverride=true&noPropertyAccessFromIndexSignature=true&ts=5.1.0-dev.20230427&filetype=d.ts#code/JYWwDg9gTgLgBAbzgBShA5lAhiOBfOAMzVwHIYBPMAUwGcBjKYMGUuAendNQ2xDeC04AE2r0ANlijVhcAEYBXeMBhCAblnELqcQXAB21NdShxpWYQDoAFADYAjAGZHASgCwAKE+iJUnSAhhBXEdcio6RmZWRE84OLhqAA9IWDh6CH1aeEIICAAuFDRMHE88IA)

### 💻 Code

```ts
import { Program } from 'typescript'

declare module 'typescript' {
    export const foo: Program
}
```

### 🙁 Actual behavior

```
'Program' is declared but its value is never read. (6133)
```

### 🙂 Expected behavior
no warning

**Comment by DetachHead** [comment]
@RyanCavanaugh i thought that too, but removing the import caused an error:
```ts
declare module 'typescript' {
    export const foo: Program // Cannot find name 'Program'. (2304)
}
```

but playing around with it more i noticed that adding an `export {}` fixes it, so i guess this just another weird thing that non-module files do?
